Projects/My_website/myWebsite/blog/signals.py

```diff
@@ -0,0 +1,21 @@
+from django import dispatch
+from django.dispatch import receiver
+from django.contrib.auth.models import User
+from blog.models import Post
+from django.db.models.signals import pre_delete, post_save, post_delete
+from django.contrib.auth.signals import user_logged_in, user_logged_out
+
+
+@receiver(post_save, sender=User)
+def action_user_login(sender, instance, **kwargs):
```

Comment:
@codal-uchoudhary any idea what does ** kwargs do here?

What all fields are there in it?

Reply:
**kwargs :- in receiver function we have to take  ' **kwargs ' parameter ,That’s because at any point arguments could get added to the signal and our receiver must be able to handle those new arguments. but in this **kwargs contains nothing.

sender :- sender is the object from which we receive the signals.

instance :- instance of the model which is being save or delete.